pkg/consensus/consensus.go

```diff
@@ -328,7 +331,19 @@ func (s *service) getBlock(h util.Uint256) block.Block {
 
 func (s *service) getVerifiedTx(count int) []block.Transaction {
 	pool := s.Config.Chain.GetMemPool()
-	txx := pool.GetVerifiedTransactions()
+	txx := make([]*transaction.Transaction, 0, len(s.lastProposal))
+
+	if s.dbft.ViewNumber > 0 {
+		for i := range s.lastProposal {
+			if tx, ok := pool.TryGetValue(s.lastProposal[i]); ok {
+				txx = append(txx, tx)
+			}
+		}
+	}
+
+	if len(txx) <= len(s.lastProposal)/2 {
```

Comment:
I'd prefer to keep this logic under `s.dbft.ViewNumber > 0`, it makes no sense comparing these when `ViewNumber == 0`. Or maybe do `len(txx) == 0 ||`. I see that technically `0 <= len(s.lastProposal)/2` is always true, but we can make code a little more readable here.